File: Exception/CustomException.py
class ValidationError(Exception):
    def __init__(self, message, field):
        super().__init__(message)
        self.field = field


class NetworkError(Exception):
    def __init__(self, message, status_code):
        super().__init__(message)
        self.status_code = status_code


def validate_age(age):
    if not isinstance(age, (int, float)):
        raise ValidationError("Age must be a number", "age")
    if age < 0 or age > 150:
        raise ValidationError("Age out of range", "age")
    return True


try:
    validate_age("old")
except Exception as e:
    if isinstance(e, ValidationError):
        print(f"Validation failed on '{e.field}': {e}")
    else:
        raise

File: Exception/CustomException.js
class ValidationError extends Error {
  constructor(message, field) {
    super(message);
    this.name = "ValidationError";
    this.field = field;
  }
}

class NetworkError extends Error {
  constructor(message, statusCode) {
    super(message);
    this.name = "NetworkError";
    this.statusCode = statusCode;
  }
}

function validateAge(age) {
  if (typeof age !== "number")  throw new ValidationError("Age must be a number", "age");
  if (age < 0 || age > 150)    throw new ValidationError("Age out of range", "age");
  return true;
}

try {
  validateAge("old");
} catch (e) {
  if (e instanceof ValidationError) {
    console.log(`Validation failed on '${e.field}': ${e.message}`);
  } else {
    throw e; // re-throw unknown errors
  }
}